experiments/add_sf_to_df_add_row.R

```diff
@@ -169,10 +175,29 @@ geo_attributes <- function(dat, zoomto = NULL){
 
                   updateTextInput(session, 'new_name', value = NA)
                   showNotification('Added New Column')
-
+      }
                   })
 
-    observeEvent(input$row_add, {
+
+    #create a vector input for 'row_add'
+    EVT_ADD_ROW <- "row_add"
+
+    # determines whether to use 'row_add' or 'map_draw_feature'
+    # also, if rows are selected then it won't trigger the 'map_draw_feature'
+
+    addRowOrDrawObserve <- function(event, id) {
```

Comment:
I like this idea, so essentially a new row can be added by the form or by the map?

Reply:
yup. it's just like adding 'new row' when 'new row' text is blank. If textInput is populated and you use 'map'  it **won't** add textInput.... Not a huge deal.

Reply:
After thinking about this, I'll mess around and see if there is a simple way to get textInput and 'map_draw' linked up. Reason 1), if you want to replicate a 'id' a bunch without having to retype this would be important, e.g. remote sensing (fire or no-fire), 2.) it's what you would expect as a user.